timestamp-impl/src/main/java/com/palantir/timestamp/TimestampAllocationFailures.java

```diff
@@ -32,6 +32,7 @@
     private final Logger log;
 
     private volatile Throwable previousAllocationFailure;
+    private volatile boolean encounteredMultipleRunningTimestamps = false;
```

Comment:
Just a thought: do we need `@Contended` here? I'd like to try the benchmarks with that added to see if it makes a difference 🙂